mqtt/mqtt3/src/proto/packet.rs

```diff
@@ -170,11 +185,6 @@ impl PacketMeta for Connect {
         }
 
         let protocol_level = src.try_get_u8()?;
-        if protocol_level != crate::PROTOCOL_LEVEL {
```

Comment:
While testing, I found a bug that violated [MQTT-3.1.2-2].
```
The Server MUST respond to the CONNECT Packet with a CONNACK return code 0x01 (unacceptable protocol level) and then disconnect the Client if the Protocol Level is not supported by the Server [MQTT-3.1.2-2].
```

Reply:
You just removed this check? Could you please point me to a unit test that covers this case?

Reply:
Test named `wrong_protocol_version_rejected` that found it.